### PR TITLE
mcp: improve promote fixture test and promote api corner cases

### PR DESCRIPTION
**Pull request by yuwenma**
### Change description

Fixes #

#### Does this PR add something which needs to be 'release noted'?

```release-note

```

- [ ] Reviewer reviewed release note.

#### Additional documentation e.g., references, usage docs, etc.:

```docs

```

#### Intended Milestone

Please indicate the intended milestone. 
- [ ] Reviewer tagged PR with the actual milestone.

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: experiments/mcp/src/pkg/promotion.py [hunk 1]
```diff
@@ -270,11 +270,19 @@ def promote_test_fixture(test_fixture_path: str, target_version: str, base_dir:
         source_version = get_version_from_path(abs_test_fixture_path)
         new_test_fixture_path_rel = test_fixture_path.replace(source_version, target_version)
         new_test_fixture_path_abs = to_abs_path(new_test_fixture_path_rel, base_dir)
+        # Move the dir instead of copying to avoid issues with existing directories
+        if os.path.exists(new_test_fixture_path_abs):
+            shutil.rmtree(new_test_fixture_path_abs)
```
Comment (yuwenma) [inline experiments/mcp/src/pkg/promotion.py:275]: This will get rid of the "#1" in .goldn.yaml files. It happens when WRITE_GOLDEN_OUTPUT find 2 tests of the same name (One for alpha and one for beta)

File: experiments/llm-eval/evaluator.py [hunk 2]
```diff
@@ -136,7 +136,31 @@ def run_test_case(self, name, prompt, verifier_script=None, cleanup_script=None,
...
+                command = ["bash", verifier_script_path]
+                process = subprocess.run(
+                    command,
+                    capture_output=True,
+                    text=True,
+                    check=False,
+                    cwd=verifier_run_dir,
+                )
+                success = process.returncode == 0
+                verifier_stdout = process.stdout
```
Comment (justinsb) [inline experiments/llm-eval/evaluator.py:157]: I wonder if this is what is eating your quota!

File: experiments/mcp/src/server.py [hunk 3]
```diff
@@ -275,11 +275,18 @@ async def scenario_promote(self, kind: str, targetVersion: str) -> str:
         # Prepare the instruction for Gemini
         instruction = f"""Promote the KCC resource of kind '{kind}' to version '{targetVersion}'. This is a three-step process.
 
-1.  **Promote the API**: Use the `promote_api` tool. For example: `promote_api(apiPath='{apiPath}', targetVersion='{targetVersion}')`. If the tool returns an error, analyze the error message. If the error is a compilation error, use the `promote_api_prompt` tool to get instructions on how to fix the dependencies. For example: `promote_api_prompt(apiPath='{apiPath}', targetVersion='{targetVersion}')`. After fixing the dependencies, retry the `promote_api` tool. Once successful, proceed to the next step.
+1.  **Promote the API**: Use the `promote_api` tool. For example: `promote_api(apiPath='{apiPath}', targetVersion='{targetVersion}')`.
```
Comment (justinsb) [inline experiments/mcp/src/server.py:278]: You may want to revisit the linked-doc approach ... this looks like it's equivalent to a doc.

**Review by justinsb — APPROVED** [review]
